### ISSUE TITLE
Proxy EDGE is not being reflected in the post_logout_redirect_uri - Admin Console Logut button

### ISSUE THREAD

**Issue by luneo7** [keycloak/keycloak#11580]
### Describe the bug

When trying to logout using the old account page the logout link is appending a `post_logout_redirect_uri` that uses the schema and port of the running container instead of the proxy one.
For instance, Keycloak is running with:
``` 
ENV KC_HTTP_ENABLED=true
ENV KC_PROXY=edge
ENV KC_HOSTNAME_STRICT=false
ENV KC_HOSTNAME_STRICT_HTTPS=false
```
All the usual endpoints and calls works, but the logout button that fetches the logout URL from `services/src/main/java/org/keycloak/forms/account/freemarker/model/UrlBean.java` and that URL is coming with plain HTTP schema and 8080 port, even though we have AWS API GW forwarding and doing TLS termination, so instead of building the URL as `https://keycloakUrl.mydomain.com` it is building as `http://keycloakUrl.mydomain.com:8080` 

### Version

18.0.0

### Expected behavior

Keycloak should build the logout URL using the host and port coming from the forwaded headers.

### Actual behavior

Keycloak is using the container server schema and port to build the logout URL

**Comment by luneo7** [comment]
Found the issue...
We have two ways of accessing Keycloak... one to hit public endpoints, through AWS API GW (that hits an ALB internally through a VPC link), and to hit internal endpoints that hits directly the app ALB.
When the traffic comes through AWS API GW it uses the `forwarded` header, here is a sample of the headers:
```
{
    "Cookie": "...",
    "content-length": "0",
    "User-Agent": "PostmanRuntime/7.29.0",
    "X-Forwarded-Proto": "http",
    "Host": "api.myhost.com:8508",
    "X-Forwarded-Port": "8508",
    "forwarded": "for=10.122.120.105;host=api.myhost.com;proto=https",
    "via": "HTTP/1.1 AmazonAPIGateway",
    "accept": "*/*",
    "X-Forwarded-For": "10.01.13.28",
    "Postman-Token": "a3483851-fd39-41bd-913f-7aeacdab26bb",
    "cache-control": "no-cache",
    "accept-encoding": "gzip, deflate, br"
}
```
And when we hit it directly through the ALB we have headers like these:
```
{
    "Cookie": "...",
    "Cache-Control": "no-cache",
    "Accept": "*/*",
    "X-Forwarded-Proto": "http",
    "User-Agent": "PostmanRuntime/7.29.0",
    "X-Forwarded-For": "10.122.120.105",
    "Host": "keycloak.app.internal.myhost.com",
    "Postman-Token": "18b23a0f-6ab5-4610-8b0d-fd981b88d7d1",
    "Accept-Encoding": "gzip, deflate, br",
    "X-Forwarded-Port": "80",

}
```

So Keycloak is not honouring the `forwarded` header and is just using the `x-forwarded` headers... I took a look at Quarkus ForwardedParser (https://github.com/quarkusio/quarkus/blob/2.7.5.Final/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java)... it doesn't work as stated in Quarkus docs (https://quarkus.io/guides/http-reference#reverse-proxy):
> Both configurations related to standard and non-standard headers can be combined, although the standard headers configuration will have precedence.

Instead if you enable the `allowForwarded` option it disregards the `x-forwarded` and there is no way of having them both working together... so I think we will either need a custom proxy handler and add that to Quarkus Vert.x config somehow... or Quarkus will need to change this and enable the defacto `forwarded` header with a fall back to `x-forwarded`...

In order to fix this, internally, I had to patch quarkus-vertx-http.... so I compile locally a version of `ForwardedParser` that accepted both forwarded headers, and updated the Keycloak jar file with `zip -u io.quarkus.quarkus-vertx-http-2.7.5.Final.jar io/quarkus/vertx/http/runtime/ForwardedParser.class` when doing the docker image build

**Comment by mposolda** [comment]
@luneo7 Question: Did you configure hostname as specified in the documentation https://www.keycloak.org/docs/latest/server_installation/#_hostname ?

**Comment by luneo7** [comment]
It was a issue with Quarkus, I patched it manually to get it running, but I've also opened an issue in Quarkus and discussed with them, a PR was made and I was fixed, the fix will be out in Quarkus 2.10 (https://github.com/quarkusio/quarkus/pull/25117)

**Comment by luneo7** [comment]
What will be missing in Keycloak is to support both allow x forwarded and forwarded headers at the same time, since the Keycloak config translator just enables one at time in Quarkus, easy to circumvent though.

**Comment by mposolda** [comment]
I am adding this with the label "Help wanted" as Keycloak team probably won't have time to look at this in the near future. Anyone is welcome to investigate this further and eventually send PR for this (if it was not automatically fixed by update to Quarkus 2.10)

**Comment by pedroigor** [comment]
@luneo7 Could you please check https://github.com/keycloak/keycloak/pull/17136.

Thanks for the investigation and fix in Quarkus.
